Source/WebKit/UIProcess/ios/WKMouseGestureRecognizer.mm

```diff
@@ -90,8 +92,12 @@ - (BOOL)_shouldReceiveTouch:(UITouch *)touch forEvent:(UIEvent *)event recognize
     return touch == _currentTouch && touch._isPointerTouch;
 }
 
-- (WebKit::NativeWebMouseEvent *)lastMouseEvent
+- (WebKit::NativeWebMouseEvent *)prepareToDispatchMouseEvent
 {
+    if (!_lastEvent || (_lastEvent->type() == WebKit::WebEvent::MouseDown && _lastDispatchedEventIsMouseDown))
+        return nullptr;
+
+    _lastDispatchedEventIsMouseDown = _lastEvent->type() == WebKit::WebEvent::MouseDown;
```

Comment:
should we also compare the `timestamp` and/or `position` just to make sure we're not somehow ruining a scenario where there are two `"mousedown"` separated by time/position in a row?

Reply:
What scenario do you have in mind?

Reply:
i dunno maybe some sort of "mousedown, switch apps, mouseup, switch back, mousedown"?

i'm not saying that this exists today, but i'd hate for this logic to cause us issues in the future if it does suddenly become possible

Reply:
I see. I'll double check that scenario, but I would expect to see a cancel event which should make this work. I also don't think checking timestamp or location is quite the right way to solve this problem, either, since both are somewhat arbitrary.

One thing I can do here is go for the other approach I described in the commit message, where we just take the last mouse event. (This results in a cleaner implementation as well, but I opted to not go for this because I deemed it slightly riskier).